src/components/contextpopup/ContextPopupDirective.js

```diff
@@ -62,6 +65,23 @@ goog.require('ga_permalink');
               return coord + ' ' + zone;
             };
 
+            var updateW3W = function() {
```

Comment:
it seems to me that this function should be in a `gaW3W` service to make it testable and reusable.

I've seen you create a `gaSearchW3WAnalyser` which could be the same.